access_custom_page.go

```diff
@@ -0,0 +1,104 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type AccessCustomPageType string
+
+const (
+	Forbidden      AccessCustomPageType = "forbidden"
+	IdentityDenied AccessCustomPageType = "identity_denied"
+)
+
+type AccessCustomPage struct {
+	// The HTML content of the custom page.
+	CustomHTML string               `json:"custom_html,omitempty"`
+	Name       string               `json:"name,omitempty"`
+	AppCount   int                  `json:"app_count,omitempty"`
+	Type       AccessCustomPageType `json:"type,omitempty"`
+	UID        string               `json:"uid,omitempty"`
+}
+
+type AccessCustomPageListResponse struct {
+	Response
+	Result     []AccessCustomPage `json:"result"`
+	ResultInfo `json:"result_info"`
+}
+
+type AccessCustomPageResponse struct {
+	Response
+	Result AccessCustomPage `json:"result"`
+}
+
+func (api *API) AccessCustomPages(ctx context.Context, rc *ResourceContainer, pageOpts PaginationOptions) ([]AccessCustomPage, error) {
+	uri := buildURI(fmt.Sprintf("/%s/%s/access/custom_pages", rc.Level, rc.Identifier), pageOpts)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return []AccessCustomPage{}, err
+	}
+
+	var customPagesResponse AccessCustomPageListResponse
+	err = json.Unmarshal(res, &customPagesResponse)
+	if err != nil {
+		return []AccessCustomPage{}, err
+	}
+	return customPagesResponse.Result, nil
+}
+
+func (api *API) AccessCustomPage(ctx context.Context, rc *ResourceContainer, customPageID string) (AccessCustomPage, error) {
+	uri := fmt.Sprintf("/%s/%s/access/custom_pages/%s", rc.Level, rc.Identifier, customPageID)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return AccessCustomPage{}, err
+	}
+
+	var customPageResponse AccessCustomPageResponse
+	err = json.Unmarshal(res, &customPageResponse)
+	if err != nil {
+		return AccessCustomPage{}, err
+	}
+	return customPageResponse.Result, nil
+}
+
+func (api *API) CreateAccessCustomPage(ctx context.Context, rc *ResourceContainer, customPage AccessCustomPage) (AccessCustomPage, error) {
```

Comment:
from the experimental client docs, the third parameter is generally called `params`.

```suggestion
func (api *API) CreateAccessCustomPage(ctx context.Context, rc *ResourceContainer, params AccessCustomPage) (AccessCustomPage, error) {
```

additionally, it should accept `CreateAccessCustomPage` params which may be a duplicate struct of `AccessCustomPage` but only with the parameters required for creation.